### PR TITLE
docs: update

### PR DESCRIPTION
**Pull request by elkowar**
# Pull request

##### Fill either one
- [ ] Does your pull request add a crate?
- [ ] Does your pull request add a new language?

- [x] Is your pull request memory safe?
- [ ] Is your pull request configurable?
- [ ] Is your pull request minimal?
- [ ] Is your pull request blazing fast?

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -105,22 +160,27 @@ unsafe fn u8(u8: u8) {
...
+    });
     assert!(!val);
 }
 
 unsafe fn punch_card() -> impl std::fmt::Debug {
-        ..=..=.. ..    .. .. .. ..    .. .. .. ..    .. ..=.. ..
+    ..=..=.. ..    .. .. .. ..    .. .. .. ..    .. ..=.. ..
         ..=.. ..=..    .. .. .. ..    .. .. .. ..    ..=..=..=..
         ..=.. ..=..    ..=.. ..=..    .. ..=..=..    .. ..=.. ..
         ..=..=.. ..    ..=.. ..=..    ..=.. .. ..    .. ..=.. ..
```
Comment (legendofmiracles) [inline src/main.rs:186]: This introduces breaking changes

**Comment by legendofmiracles** [comment]
Makes too little use of nightly features 👎

File: src/main.rs (hunk 1 above)
Comment (mTvare6) [reply src/main.rs:186]: Yes, please fix asap, I don't want enterprise users to stop using

**Review by legendofmiracles — APPROVED** [review]
Otherwise LGTM!

File: src/main.rs [hunk 2]
```diff
@@ -8,45 +8,77 @@
...
+use r_i18n::I18nConfig;
 
-unsafe fn strange() -> bool { let _x: bool = return true; }
+macro_rules! make_function {
+    (name = $name:ident, args = $($arg:ident : $argtype:ty),*, return_type = $ret_type:ty, body = $body:block) => {
+        unsafe fn $name ($($arg: $argtype),*) -> $ret_type $body
+    }
+}
 
-unsafe fn funny() {
```
Comment (dylanaraps) [inline src/main.rs:16]: bye bye funny :(

Comment (legendofmiracles) [reply src/main.rs:16]: :(

File: src/main.rs (hunk 1 above)
Comment (elkowar) [reply src/main.rs:186]: hmmm, what about just bumping the major version? keeping backwards compatibility here would add significant maintainance overhead :/

Comment (legendofmiracles) [reply src/main.rs:186]: Absolutely out of question, this is *the* major feature for hello-world.rs if it wouldn't be for this we'd lose all revenue and customers.

(fr tho, look at this: https://discord.com/channels/273534239310479360/274215136414400513/875935651177050162)

**Comment by mTvare6** [comment]
Thanks, now we can replace all bad unsafe c programs like `echo` with very minimal, blazing fast, and configurable `hello-world`

**Pull request merged**